aikau/src/main/resources/alfresco/forms/controls/Tree.js

```diff
@@ -0,0 +1,179 @@
+         return this.createTree();
+      },
+
+      /**
+       * This creates a new [tree]{@link module:alfresco/navigation/Tree}.
+       * 
+       * @instance
+       * @return {object} A new tree instance
+       */
+      createTree: function alfresco_forms_controls_Tree__createTree() {
+         return new Tree({
+            pubSubScope: this.pubSubScope,
+            publishTopic: this._treeNodeSelectedTopic,
+            childRequestPublishTopic: this.optionsConfig.publishTopic,
+            childRequestPublishPayload: this.optionsConfig.publishPayload,
+            childRequestPublishGlobal: this.optionsConfig.publishGlobal
+         });
+      },
+
+      /**
+       * Overrides the [inherited function]{@link module:alfresco/forms/controls/BaseFormControl#getPubSubOptions}
+       * to destroy any existing tree and then re-create it. This is necessary because new options effectively mean
+       * a new tree. The tree is only destroyed if it already exists to accomodate the form control creation workflow.
+       *
+       * @instance
+       * @param {object} config The options config (which is ignored)
+       */
+      getPubSubOptions: function alfresco_forms_controls_Tree__getPubSubOptions(/*jshint unused:false*/ config) {
+         if (this.wrappedWidget)
+         {
+            this.wrappedWidget.destroy();
+            this.wrappedWidget = this.createTree();
+            this.placeWidget();
+         }
+      },
+
+      /**
+       * This function is called whenever the user selects a node in the tree. It calls 
+       * [onValueChangeEvent]{@link module:alfresco/forms/controls/BaseFormControl#onValueChangeEvent}
+       * to ensure that the form value is updated and can be validated (and potentially
+       * trigger changes in other fields).
+       * 
+       * @instance
+       * @param {object} payload The selected tree node value.
+       */
+      onTreeNodeSelected: function alfresco_forms_controls_Tree__onTreeNodeSelected(payload) {
+         this.alfLog("info", "Tree node selected", payload);
+         var value = payload;
+         if (this.valueProperty)
+         {
+            value = lang.getObject(this.valueProperty, false, payload);
+         }
+         this.onValueChangeEvent(this.name, this.value, value);
+         this.value = value;
+      },
+      
+      /**
+       * Gets the value of the currently selected node in the tree.
+       * 
+       * @instance
+       * @returns {boolean} The checked state of the checkbox
+       */
+      getValue: function alfresco_forms_controls_Tree__getValue() {
+         return this.value;
+      },
+      
+      /**
+       * Currently it is not possible to set a tree node value.
+       * 
+       * @instance
+       * @param {object} value The value to set.
+       */
+      setValue: function alfresco_forms_controls_Tree__setValue(value) {
+         // jshint unused:false
+         // Not possible to set a value currently
```

Comment:
Feel we should at least log this when someone tries to do it, if not throwing an error?